templates/shared/forms/_form-validation.html

```diff
@@ -0,0 +1,175 @@
+
+      // Submit the form in the background
+      backgroundSubmit(marketoForm);
+    }
+  }
+
+  backgroundSubmit = function(marketoForm, submitCallback) {
+    var request = new XMLHttpRequest();
+    var submitUrl = marketoForm.getAttribute('action');
+    var formData = serialize(marketoForm);
+
+    request.open("POST", submitUrl, true);
+
+    //Send the proper header information along with the request
+    request.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+
+    // When request has finished, call the callback function
+    if (submitCallback) {
+      request.addEventListener(
+      'readystatechange',
+      function() {
+        if (this.readyState == 4) {
+          // Pass context and arguments on to submitCallback
+          submitCallback.apply(this, arguments);
+        }
+      });
+    }
+
+    // recaptcha test
+    var recaptchaListElement = marketoForm.getElementsByClassName('g-recaptcha')[0];
+    var recaptchaWidgetId = recaptchaListElement.dataset.widgetId;
+    var response = grecaptcha.getResponse(recaptchaWidgetId);
+    var already_errored = document.getElementById('recaptcha-msg');
+    if (response === '') {
+      if (!already_errored) {
+        recaptchaListElement.classList.add("recaptcha-is-error");
+        recaptchaListElement.insertAdjacentHTML('afterend', '<p id="recaptcha-msg" class="p-form-validation__message" role="alert"><strong>Error:</strong> You need to complete the recaptcha to submit this form. </p>');
+      }
+      return false;
+    } else {
+      if (already_errored) {
+        recaptchaListElement.classList.remove("recaptcha-is-error");
+        var msg = document.getElementById('recaptcha-msg');
+        msg.parentNode.removeChild(msg);
+      }
+    }
+
+    // copy recaptcha to marketo field
+    var formData = formData.concat("&grecaptcharesponse=" + response);
+
+    // Send off the POST request
+    request.send(formData);
+
+    // get the download asset if it exists
+    var download_asset_url = marketoForm.querySelector('input[name=download_asset_url]');
+    if (download_asset_url != null) {
+      download_asset_url = download_asset_url.value;
+    }
+
+    // get the return url if it exists to redirect users after
+    var return_url = marketoForm.querySelector('input[name=return_url]');
+    if (return_url != null) {
+      return_url = return_url.value;
+    }
+
+    // check if it is a dynamic modal form
+    var isModal = marketoForm.classList.contains("modal-form");
+
+    // check if there is a thank you message to post
+    var thankYouMessage = marketoForm.querySelector('input[name=thankyoumessage]');
+    if (thankYouMessage != null) {
+      thankYouMessage = thankYouMessage.value;
+    }
+    formid = this.id;
+
+    console.log("download_asset_url: " + download_asset_url);
+    console.log("return_url: " + return_url);
+    console.log("modal: " + isModal);
+    console.log("thankYouMessage: " + thankYouMessage);
```

Comment:
Can you remove these console logs please

Reply:
done